R/Step2_QUAIL_vQTL.R

```diff
@@ -0,0 +1,199 @@
+#' Perform Univariate Regression for Genotype Data
+#'
+#' @param X Numeric matrix of genotypes (n x p), where n is the number of samples and p is the number of SNPs.
+#' @param y Numeric vector of phenotypes (length n).
+#' @param Z Optional numeric matrix of covariates (n x k), where k is the number of covariates.
+#' @param center Logical, whether to center the data (default: TRUE).
+#' @param scale Logical, whether to scale the data (default: FALSE).
+#' @param return_residuals Logical, whether to return residuals (default: FALSE).
+#' @return A list containing regression results: \code{betahat}, \code{sebetahat}, \code{z_scores}, \code{p_values}, and \code{q_values}.
+#' @examples
+#' X <- matrix(rnorm(1000), 100, 10)
+#' y <- rnorm(100)
+#' results <- univariate_regression(X, y)
+
+univariate_regression = function (X, y, Z = NULL, center = TRUE,
```

Comment:
Since in susieR there is a function of the same name and is exported, let's not export it here. Please make it an internal function I think in roxgen2 it's just like `@keyword internal` or something so it's not going to be available via pecotmr::univariate_regression